desc/compute/_surface.py

```diff
@@ -217,6 +221,9 @@ def _phi_z_Surface(params, transforms, profiles, data, **kwargs):
     parameterization="desc.geometry.core.Surface",
 )
 def _Z_Surface(params, transforms, profiles, data, **kwargs):
+    errorif(
+        kwargs.get("basis", "rpz").lower() not in {"rpz", "xyz"}, NotImplementedError
```

Comment:
We can instead just add this line with PR #1027 to the main `compute` function. This shouldn't be added to every single compute function one by one.